Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -424,7 +433,7 @@ await taskTable.RemoveRemainingDataDependenciesAsync(dependentTasks,
     // Find all tasks whose dependencies are now complete in order to start them.
     // Multiple agents can see the same task as ready and will try to start it multiple times.
     // This is benign as it will be handled during dequeue with message deduplication.
-    var readyTasks = await taskTable.FindTasksAsync(data => dependentTasks.Contains(data.TaskId) && data.Status == TaskStatus.Creating &&
+    var readyTasks = await taskTable.FindTasksAsync(data => dependentTasks.Contains(data.TaskId) && data.Status == TaskStatus.Pending &&
```

Comment:
I wonder if you should keep the test on Creating as well.

Reply:
We do not really need it. Except for two things:
- if old messages are in the queue (should not happen much)
- if we want to save a request that changes the task status to Pending when we know it will be changed to Submitted shortly after